bleep-core/src/scala/bleep/GenBloopFiles.scala

```diff
@@ -328,6 +328,27 @@ object GenBloopFiles {
         case _                                            => bloop.config.Tag.Library
       }
 
+    val sourceGenerators = {
+      val sg = explodedProject.`source-generator-scripts`.values.map { sc =>
+        if (!build.scripts.contains(sc.scriptName)) {
+          sys.error(s"Unknown script ${sc.scriptName} referenced by ${crossName.value}")
+        }
+        val codegenPath = buildPaths.generatedSourcesDir(crossName)
```

Comment:
let's drop the arguments thing in this first version. if we do it (and I suspect we won't have to) we'll do it across the board in a later PR.

Reply:
I'd rather not drop it, at least not without a way to communicate to the script what module it was triggered for (unless there's a way to achieve it without args, which I doubt there is). 

What I have codified in this draft is that "$this" passed in an arg list would be a way to do just that.

Reply:
Is the reason why you want this that you're thinking of cross projects where you need to generate the same resources many times?

Reply:
Ok, I've been thinking ten minutes. Let's keep the functionality of passing project name as parameter.

Just to explain my somewhat negative position:
- passing arguments feels like programming in YAML, which is not a favourite pastime of mine.
- I'd rather that we make each script each own entry in `scripts` and a corresponding scala `object`. super clear and obvious, easy to run, easy to debug.

However, since the bloop resource generators are connected to a bloop project, we really lose the opportunity to run/track the resource generator for all `CrossId`s in one go. That we'll just have to live with.

How about we just establish the convention that all resource generators are passed the project name as a parameter? No configuration or flexibility, just that?

Reply:
Well an example of why I want that is: how do you determine the target folder if you don't know the module which invokes the code generator ? Or, what if what you want to generate is dependant on the scala version of the module in question ?

Reply:
> How about we just establish the convention that all resource generators are passed the project name as a parameter? 

Would that mean having a `GeneratingScript` class inherting the `Script` class ? That might work (from a DX perspective, that is)

> passing arguments feels like programming in YAML, which is not a favourite pastime of mine.

Yeah I don't disagree there. 

> I'd rather that we make each script each own entry in scripts and a corresponding scala object. super clear and obvious, easy to run, easy to debug.

> No configuration or flexibility, just that?

This is a little too strict for my taste, I do think it'd be interesting to allow for users to pass some configuration to the scripts in the `bleep.yaml`. Say someone wants to publish pre-built scripts as literal "plugins" : this would allow people to customise the behaviour of the plugin without having to touch 2 files (the build file to pull the dependency and configure a script project + a scala file). 

To elaborate : if you have a `metadata` field in the projects configuration that'd receive an arbitrary Json object, bleep could pass that object to scripts and script implementors could use that to retrieve configuration elements however they please. If users would rather duplicate some Scala files instead of using yaml configuration, they'd totally be able to do that. What's more : you could deep-merge the `metadata` value of all the templates (and the one of the project itself) before passing it to the script. Imho, this would be coherent with the way bleep is configured (declaratively), generally speaking, and would prevent having to compile scala code when reconfiguring some aspects of the build (which is a one of the appeals of bleep, as having to compile a Scala-written build is detrimental to the snappiness of the build's commands)./

Reply:
> Would that mean having a GeneratingScript class inherting the Script class ? That might work (from a DX perspective, that is)
Yes - that seems fitting! :)

> This is a little too strict for my taste, I do think it'd be interesting to allow for users to pass some configuration to the scripts in the bleep.yaml. Say someone wants to publish pre-built scripts as literal "plugins" : this would allow people to customise the behaviour of the plugin without having to touch 2 files (the build file to pull the dependency and configure a script project + a scala file).

My immediate thought is that bleep has not one but three main goals:
1) as simple as possible
2) as fast as possible
3) automatic import from sbt

What you're suggesting here works towards 2), but against 1). I actually think 1) may be more important, and "one way to do it" is hugely important for 1). That way is to write scala scripts.

Anyways, the merging of metadata fields appeals to me. Give me a day or two to chew on this :)

Reply:
I lifted this conversation to https://github.com/oyvindberg/bleep/issues/272 and will close the thread